Tiltfile

```diff
@@ -0,0 +1,201 @@
+# -*- mode: Python -*-
+
+k8s_yaml([
+'config/crd/bases/velero.io_backups.yaml',
+'config/crd/bases/velero.io_backupstoragelocations.yaml', 
+'config/crd/bases/velero.io_deletebackuprequests.yaml',   
+'config/crd/bases/velero.io_downloadrequests.yaml',   
+'config/crd/bases/velero.io_podvolumebackups.yaml',  
+'config/crd/bases/velero.io_podvolumerestores.yaml',  
+'config/crd/bases/velero.io_resticrepositories.yaml',
+'config/crd/bases/velero.io_restores.yaml',    
+'config/crd/bases/velero.io_schedules.yaml', 
+'config/crd/bases/velero.io_serverstatusrequests.yaml',  
+'config/crd/bases/velero.io_volumesnapshotlocations.yaml', 
+])
+
+# default values
+settings = {
+   "default_registry": "",
+   "enable_restic": False,
+   "create_backup_locations": False, 
+   "setup-minio": False, 
+   "local_goos": "darwin",
+}
+
+# global settings
+settings.update(read_json(
+    "tilt-resources/tilt-settings.json",
+    default = {},
+))
+
+k8s_yaml(kustomize('tilt-resources'))
+k8s_yaml('tilt-resources/deployment.yaml')
+if settings.get("enable_restic"):
+    k8s_yaml('tilt-resources/restic.yaml')
+if settings.get("create_backup_locations"):
+    k8s_yaml('tilt-resources/velero_v1_backupstoragelocation.yaml')
+if settings.get("setup-minio"):
+    k8s_yaml('examples/minio/00-minio-deployment.yaml', allow_duplicates=True)
+
+# By default, Tilt automatically allows Minikube, Docker for Desktop, Microk8s, Red Hat CodeReady Containers, Kind, K3D, and Krucible.
+allow_k8s_contexts(settings.get("allowed_contexts"))
+default_registry(settings.get("default_registry"))
+local_goos = settings.get("local_goos", )
```

Comment:
I think we could use [`local`](https://docs.tilt.dev/api.html#api.local) here to call `go env GOOS` which would eliminate the need to explicitly state the value in settings. I tried out the following and it worked for me on my linux box:
`local_goos = str(local("go env GOOS", quiet=True, echo_off=True)).strip()`